src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,161 @@
+     */
+    reflectionFormatSRGBA8?: boolean;
+}
+
+/**
+ * Light Estimation Feature
+ *
+ * @since 5.0.0
+ */
+export class WebXRLightEstimation extends WebXRAbstractFeature {
+
+    private _reflectionCubeMap: Nullable<WebGLTexture> = null;
+    private _xrLightEstimate: Nullable<XRLightEstimate> = null;
+    private _xrLightProbe: Nullable<XRLightProbe> = null;
+    private _xrWebGLBinding: Nullable<XRWebGLBinding> = null;
+
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.LIGHT_ESTIMATION;
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+
+    /**
+     * This observable will notify when the reflection cube map is updated.
+     */
+    public onReflectionCubeMapUpdatedObservable: Observable<WebGLTexture> = new Observable();
+
+    /**
+    * Creates a new instance of the light estimation feature
+    * @param _xrSessionManager an instance of WebXRSessionManager
+    * @param options options to use when constructing this feature
+    */
+    constructor(
+        _xrSessionManager: WebXRSessionManager,
+        /**
+         * options to use when constructing this feature
+         */
+        public readonly options: IWebXRLightEstimationOptions
+    ) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "light-estimation";
+
+        // https://immersive-web.github.io/lighting-estimation/
+        Tools.Warn("light-estimation is an experimental and unstable feature.");
+    }
+
+    /**
+     * While the estimated cube map is expected to update over time to better reflect the user's environment as they move around those changes are unlikely to happen with every XRFrame.
+     * Since creating and processing the cube map is potentially expensive, especially if mip maps are needed, you can listen to the onReflectionCubeMapUpdatedObservable to determine
+     * when it has been updated.
+     */
+    public get reflectionCubeMap(): Nullable<WebGLTexture> {
+        return this._reflectionCubeMap;
+    }
+
+    /**
+     * The XRLightProbe object created during attach (may not be available for a few frames depending on device).
+     *
+     * The XRLightProbe itself contains no lighting values, but is used to retrieve the current lighting state with each XRFrame.
+     * XR Light Probe also contains the probe space, may update it's pose over time as the user moves around their environment.
+     */
+    public get xrLightProbe(): Nullable<XRLightProbe> {
+        return this._xrLightProbe;
+    }
+
+    /**
+     * The most recent light estimate.  Available starting on the first frame where the device provides a light probe.
+     */
+    public get xrLightingEstimate(): Nullable<XRLightEstimate> {

```

Comment:
we'll need to convert those values to left-handed when used in babylon. otherwise they are flipped.

Reply:
Yes @RaananW - To be consistent I think we want to remove XR specific typings from the API.  Likely that means wrapping with something like:
```typescript
export type LightEstimate = {
  primaryLightDirection: Vector3 // changed to left handed
  primaryLightIntensity: Vector3 // although not directly usable (see below)
  sphericalHarmonicCoefficients: Float32Array
}
```
I didn't go that extra step.  Following the explainer the `intensity` from the primary light is used to set color and a numeric primitive for light intensity (https://github.com/immersive-web/lighting-estimation/blob/main/lighting-estimation-explainer.md#xrlightestimate).